### DISCUSSION TITLE
Large amount of files in albums directory

### DISCUSSION THREAD

**Discussion by jd1** [photoprism/photoprism#2601] [Q&A] [answered]
Hi everyone,

I'm running Photoprism with ~10 000 pictures. I mappped the `storage` folder from my filesystem into the docker container(`/photoprism/storage`). Now I noticed that my filesystem is running out of inodes, because `/storage/albums` contains  more than 5 000 000 yaml files, especially `/storage/albums/folder` contains most of them. Can someone explain to me why these files are there and why I have so many of them? What happens if I delete them?

**Comment by lastzero** [comment, marked as answer]
That's unusual unless you have that many albums / folders? These are backups only, so it's safe to delete them. You can also disable YAML files in your config. See docs.

**Comment by McGeaverBeaver** [comment]
I have found this same issue, I was moving my appdata of photoprisim to new NVME storage and " albums/folder " as well has 5million + files... what could have caused this? will it happen again?

**Reply by lastzero** [reply]
As written above, it is safe to delete these backup files so you don't have to keep or move them all. I've implemented changes after learning about this problem, so hopefully it shouldn't happen again if you're using our latest release and not some old version.

**Reply by lastzero** [reply]
@McGeaverBeaver Since folders, months, states, and moments are technically also albums in PhotoPrism, one thing that could result in a lot of files being created is having a lot of subdirectories in your Originals folder.

So that could be causing it, possibly also in combination with a temporary misconfiguration where the originals folder points to `/`, or database issues that trigger repeated restarts and re-indexing.

I've just checked my personal instance (running since 2018) and found only 1317 backup files:

```
/photoprism/michael/storage/albums$ find . -type f | wc -l
1317
```

**Comment by greekdoctor** [comment]
+1 this issue. I was doing the exact same, moving the files to a new server. Noticed it was copying a serious number of files with the name a*.yaml like arvu77d1axki2u66.yml. I had 2 million of them. I don't have particularly many subdirectories in the originals folder.

Thanks to this thread, I know it's a known issue that's probably been fixed and my files are left over, and that they're safe to delete. Thanks!
